pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -190,55 +181,32 @@ public boolean readCompacted() {
      *
      * @return a SendMessageInfo object that contains the detail of what was sent to consumer
      */
-    public SendMessageInfo sendMessages(final List<Entry> entries) {
-        // Empty listener
-        return sendMessages(entries, null);
-    }
-
-    /**
-     * Dispatch a list of entries to the consumer. <br/>
-     * <b>It is also responsible to release entries data and recycle entries object.</b>
-     *
-     * @return a SendMessageInfo object that contains the detail of what was sent to consumer
-     */
-    public SendMessageInfo sendMessages(final List<Entry> entries, SendListener listener) {
+    public ChannelPromise sendMessages(final List<Entry> entries, int[] batchSizes, SendMessageInfo sendMessageInfo) {
```

Comment:
maybe rename batchSizes to threadLocalBatchSizes and same for sendMessageInfo to give a hint for future users that they can't move off the thread.

Reply:
Fixed as in "don't pass thread locals there anymore"